src/TestImages.jl

```diff
@@ -113,12 +114,60 @@ end
 
 function image_path(imagename)
     ensure_artifact_installed("images", artifacts_toml)
-    
+
     image_dir = artifact_path(artifact_hash("images", artifacts_toml))
     return joinpath(image_dir, imagename)
 end
 
 _findall(name; min_score=0.6) = findall(name, remotefiles, Winkler(Jaro()), min_score=min_score)
 _findmax(name; min_score=0.8) = findmax(name, remotefiles, Winkler(Jaro()), min_score=min_score)
 
+
+"""
+    phantom = shepp_logan(N, [M=N]; high_contrast=true)
+
+Output the NxM Shepp-Logan phantom, which is a standard test image usually used
+for comparing image reconstruction algorithms in the field of computed
+tomography (CT) and magnetic resonance imaging (MRI).
+
+If the argument `M` is omitted, the phantom is of size `NxN`. When setting the keyword argument
+`high_constrast` to false, the CT version [1] of the phantom is created. Otherwise, the high contrast
+MRI version [2] is calculated.
+
+# References
+
+[1] Shepp, Lawrence A., and Benjamin F. Logan. "The Fourier reconstruction of a head section." _IEEE Transactions on nuclear science_ 21.3 (1974): 21-43.
+[2] Toft, Peter Aundal. "The Radon transform-theory and implementation." (1996).
+"""
+function shepp_logan(N::Integer, M::Integer; high_contrast=true)
+    x = range(-1, stop=1, length=M)'
+    y = range(1, stop=-1, length=N)
+  
+    centerX = [0, 0, 0.22, -0.22, 0, 0, 0, -0.08, 0, 0.06]
+    centerY = [0, -0.0184, 0, 0, 0.35, 0.1, -0.1, -0.605, -0.605, -0.605]
+    majorAxis = [0.69, 0.6624, 0.11, 0.16, 0.21, 0.046, 0.046, 0.046, 0.023, 0.023]
+    minorAxis = [0.92, 0.874, 0.31, 0.41, 0.25, 0.046, 0.046, 0.023, 0.023, 0.046]
+    theta = [0, 0, -0.1pi, 0.1pi, 0, 0, 0, 0, 0, 0]
+  
+    if high_contrast
+        # high contrast (MRI) version of the phantom
+        grayLevel = [1, -0.8, -0.2, -0.2, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
+    else
+        # original (CT) version of the phantom
+        grayLevel = [2, -0.98, -0.02, -0.02, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
+    end
+  
+    P = zeros(N, M)
+    for l = 1:length(theta)
+      P .+= grayLevel[l] .* (
+             abs2.( (cos(theta[l]) .* (x .- centerX[l]) .+
+                sin(theta[l]) .* (y .- centerY[l])) ./ majorAxis[l] ) .+
+             abs2.( (sin(theta[l]) .* (x .- centerX[l]) .-
+                cos(theta[l]) .* (y .- centerY[l])) ./ minorAxis[l] ) .< 1 )
```

Comment:
I thought it is equivalent, but benchmark tells me that

```julia
      @. P += grayLevel[l] * (
             abs2( (cos(theta[l]) * (x - centerX[l]) +
                sin(theta[l]) * (y - centerY[l])) / majorAxis[l] ) +
             abs2( (sin(theta[l]) * (x - centerX[l]) -
                cos(theta[l]) * (y - centerY[l])) / minorAxis[l] ) < 1 )
```

is ~2x slower than the current version.

Reply:
BTW, we can avoid the reallocation.
```julia
P = zeros(Gray{Float64}, N, M)
    for l = 1:length(theta)
      P .= gray.(P) .+ grayLevel[l] .* (
...
```
Probably no slowdown, but no speedup.

Reply:
Yes, I'm aware of this. Just that I prefer to not depend on `ColorVectorSpace`.

Reply:
?
```julia
julia> using ColorTypes

julia> P = zeros(Gray{Float64}, 2, 2);

julia> P .= 1.0
2×2 Array{Gray{Float64},2} with eltype Gray{Float64}:
 Gray{Float64}(1.0)  Gray{Float64}(1.0)
 Gray{Float64}(1.0)  Gray{Float64}(1.0)
```

Reply:
😂 I was too careless that I misunderstood it!

Reply:
I know this PR isn't about optimization, but I think it's easier to read using functions.
```julia
function shepp_logan(N::Integer, M::Integer; high_contrast=true)
    x = range(-1, stop=1, length=M)'
    y = range(1, stop=-1, length=N)
  
    center_x = (0.0, 0.0, 0.22, -0.22, 0.0, 0.0, 0.0, -0.08, 0.0, 0.06)
    center_y = (0.0, -0.0184, 0.0, 0.0, 0.35, 0.1, -0.1, -0.605, -0.605, -0.605)
    major_axis = (0.69, 0.6624, 0.11, 0.16, 0.21, 0.046, 0.046, 0.046, 0.023, 0.023)
    minor_axis = (0.92, 0.874, 0.31, 0.41, 0.25, 0.046, 0.046, 0.023, 0.023, 0.046)
    theta = (0.0, 0.0, -0.1pi, 0.1pi, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0)
  
    if high_contrast
        # high contrast (MRI) version of the phantom -- [2] p.201
        gray_level = (1.0, -0.8, -0.2, -0.2, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1)
    else
        # original (CT) version of the phantom -- [1]
        gray_level = (2.0, -0.98, -0.02, -0.02, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01)
        # [3] p.439 uses the following setting, and is what MATLAB's built-in `phantom` uses
        # gray_level = (1.0, -0.98, -0.02, -0.02, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01)
    end
    
    function ellipse(dx, dy, a, b, cos_t, sin_t)
        tx = cos_t * dx + sin_t * dy
        ty = sin_t * dx - cos_t * dy
        abs2(tx * b) + abs2(ty * a) < (a * b)^2
    end
    function ellipse(dx, dy, a, b)
        abs2(dx * b) + abs2(dy * a) < (a * b)^2
    end
  
    P = zeros(Gray{Float64}, N, M)
    for l = 1:length(theta)
        dx = x .- center_x[l]
        dy = y .- center_y[l]
        if theta[l] == 0.0
            P .= gray.(P) .+ gray_level[l] .* ellipse.(dx, dy, major_axis[l], minor_axis[l])
        else
            P .= gray.(P) .+ gray_level[l] .* 
                ellipse.(dx, dy, major_axis[l], minor_axis[l], cos(theta[l]), sin(theta[l]))
        end
    end

    return P
end
```